pylint_nautobot/status_field_instead_of_status_model.py

```diff
@@ -0,0 +1,25 @@
+"""Check for deprecated StatusModel usage and encourage the usage of StatusField instead."""
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IAstroidChecker
+
+
+class NautobotDeprecatedStatusModelChecker(BaseChecker):
+    """Discourage the usage of deprecated StatusModel and encourage the usage of StatusField."""
+
+    __implements__ = IAstroidChecker
+
+    version_specifier = ">=2,<3"
+
+    name = "nautobot-status-field-instead-of-status-model"
+    msgs = {
+        "E4292": (
+            "Declare status on the model explicitly with a StatusField, as StatusModel is deprecated",
+            "nb-status-field-instead-of-status-model",
+            "https://docs.nautobot.com/projects/core/en/next/models/extras/status/#status-internals",
+        ),
+    }
+
+    def visit_classdef(self, node):
+        for basename in node.basenames:
+            if basename in ["StatusModel", "statuses.StatusModel"]:
```

Comment:
One issue I could see with this pattern is that it doesn't catch the following scenarios:
- `import StatusModel as SomethingElse`
- Somebody inheriting from a class that inherits from `StatusModel` (although I guess this would be caught on the definition of that class)

My point is, would it be better to use pylint's inference like this:
```
# node is the currently observed `ClassDef` node
ancestor_class_types = [ancestor.qname() for ancestor in node.ancestors()]
if "nautobot.extras.models.statuses.StatusModel" in ancestors_class_types:
    pass # Failure
```

The only potential issue with this is that the nautobot package needs to be installed for the inference to work properly, but I am pulling this in for #9 already.

Reply:
Good suggestion!

Reply:
Shouldn't be I think - what kind of error are you getting?

Reply:
I set a breakpoint() right after `ancestor_class_types = [ancestor.qname() for ancestor in node.ancestors()]` however the list is currently empty and the `if` statement right after wouldn't work, so `pytest` failed because it didn't see the warning message generated.
<img width="1267" alt="Screen Shot 2023-06-22 at 10 54 48 AM" src="https://github.com/nautobot/pylint-nautobot/assets/46973263/51c596d9-eb50-4ee0-aa7a-0de9a9e0a7b3">

Reply:
So I guess `node.ancestors()` is empty somehow?

Reply:
Ahhh yeah that is very unclear, sorry. The `nautobot` package needs to be installed for it to be able to infer things from it. This is already the case in #9

Reply:
#9 is now merged, if you want to update thids branch and retest.